### DISCUSSION TITLE
VCA und Guidance Steering zusammen funktionieren nicht mehr

### DISCUSSION THREAD

**Discussion by Maag000** [Mogli12/KeyboardSteer#755] [Q&A]
Set dem neusten Update von Guidance Steering V2.0.1.0 und Vehicle Control Addon V9.0.2.7 funktioniert VCA nicht mehr.
Die Tastenbelegung kann beliebig verändert werden, aber man kann im VCA das GPS nicht mehr aktivieren (STRG+W) oder auch die Breite nicht mehr einstellen. Auch wennn man die Tastenbelegung ändert, geht es nicht mehr.

**Comment by Clouseau219** [comment]
Wenn das Fahrzeug GPS hat, wird die Spurführung von VCA deaktiviert.

> Damit beide Mods sich nicht gegenseitig stören, ist die Spurführung von VCA aus, sobald das Fahrzeug GPS an Bord hat.

[https://github.com/Mogli12/VehicleControlAddon/issues/720](url)

**Comment by Mogli12** [comment]
Ich verstehe das nicht. Warum konfiguriert man GPS in ein Fahrzeug, will dann aber VCA zur Spurführung nutzen? Es ist ja überhaupt kein Problem GPS wieder aus der Konfiguration heraus zu nehmen. Wenn man es später wieder aufnimmt, dann kostet es nicht einmal mehr etwas.

**Comment by RemmyDemmy** [comment]
Hallo Mogli, erstmal vielen Dank für deine Arbeit.
Bei uns im Multiplayer ist es so, das mehrere Leute auf einem Hof spielen und der eine lieber VCA und der andere lieber GPS nutzt. Das klappt nun nicht mehr.

**Comment by Mogli12** [comment]
Da kann an doch beim Fahrzeugwechsel kurz die Konfiguration ändern, oder?

Wenn ich den Kipper erst zum Abfahren von Weizen benutzt habe, und danach damit Ballen transportieren will, dann muss ich doch auch die Konfiguration kurz ändern. 

Aber wenn das so ein Theater hier jetzt gibt, dann baue ich die Spurführung eben ganz aus.

**Comment by jason0611** [comment]
Hallo Stefan,
bitte nicht ausbauen. :-)
Ich denke, beide GPS-Varianten haben ihre Berechtigung und auch ihre Fans. Vielleicht macht es Sinn, weniger darauf zu schauen, ob Wopsters GPS konfiguriert ist, sondern eher, ob das GPS gerade aktiv ist.
Ich prüfe das in meinem Vorgewende Management über die folgenden Eigenschaften:
  ```
local spec_gs = self.spec_globalPositioningSystem 
  local gpsEnabled = spec_gs ~= nil and spec_gs.lastInputValues ~= nil and spec_gs.lastInputValues.guidanceSteeringIsActive
```
Vielleicht wäre das eine Alternative? Die Spurführung von VCA zu aktivieren, während die GS-Spurführung aktiv ist, macht natürlich in der Tat wenig Sinn.
Viele Grüße,
Martin

**Comment by Mogli12** [comment]
Wenn man die Spurführung einschalten kann, dann sind ja trotzdem Teile der Tastenbelegungen aktiv. Das sind ca. 10 Aktionen. Deshalb wird die Spurführung in VCA aktuell diaktiviert, wenn GPS verbaut ist.

**Comment by Mogli12** [comment]
Hier auf Github steht die aktuelle Entwickler-Version von VCA. Wer das nicht verstehen kann oder will, der soll VCA bitte nicht benutzen. Für mich ist Kompatibilität mit anderen Mods wichtiger als die schiere Menge an Funktionen. Ich habe auch VCA, GPS und CabView im "mods"-Ordner und versuche alles zu gut wie es geht aufeinander abzustimmen.
Ich hoffe, dass ich in einer zukünftigen Version es besser hinbekomme. Aktuell warte ich auf die Veröffentlichung der Version 1.0. Daher gibt es gerade keine Änderungen.

> Ok, schade. Dann fliegt VCA leider aus meinem Mod-Ordner.

Nicht wundern, wenn man nach solchen Kommentaren von mir blockiert wird!

**Comment by Mogli12** [comment]
P.S.: Vielen Dank an die vielen Benutzer, die mit konstruktiver Kritik helfen VCA zu verbessern!

**Comment by ghost** [comment]
Schade dass Du so reagierst und mir nicht antwortest.
Scheinst ja dann doch nicht so auf ehrliche Kritik aus zu sein.
Nichts desto trotz, danke für die bisher geleistete Arbeit.

**Comment by Mogli12** [comment]
Ich kann in "Dann fliegt VCA leider aus meinem Mod-Ordner." keine konstruktive Kritik erkennen. Ich habe jetzt mehrfach erklärt, dass ich ein Problem in den vielen Tastenbelegungen sehe und daher zunächst VCA deaktiviere, wenn GPS im Fahrzeug eingebaut ist.

**Comment by jason0611** [comment]
Hallo Stefan,
schön, dass es weitergeht. BTW: Gute Idee mit der Diskussionsfunktion. :-)
Bei der Realisierung über die Abfrage, ob die GS-Spurführung aktiv ist (über "guidanceSteeringIsActive"), kommt es leider zu einem Nebeneffekt: Immer, wenn die Spurführung aktiviert und deaktiviert wird, kommt es zu einem kurzen Hänger des Spiels. Gerade bei der Feldarbeit stört das bei jeder Wende mit der Zeit schon sehr.
Vielleicht wäre eine Lösung, die Prüfung auf "guidanceIsActive" zu ändern. Dann wird das GPS-Modul von VCA abgeschaltet, wenn der Spieler das GS-Modul selbst (nicht die Spurführung) anschaltet. Dadurch läuft die Arbeit auf dem Feld wieder flüssig, ein kurzer Hänger beim Anschalten oder Abschalten des Moduls stört nicht weiter.
Viele Grüße,
Martin

**Reply by Mogli12** [reply]
Hallo Martin,
Ich frage mich gerade, wieso es kurz hängt. Das kann eigentlich nur daran liegen, dass die Action-Bindings neu angefordert werden. 
Ich schaue mir das mal an.
LG Stefan

**Comment by Mogli12** [comment]
https://github.com/Mogli12/VehicleControlAddon/commit/4d2d2e336330b6bfa5c689d90ea398974e561926
